Userland/Libraries/LibGUI/Slider.cpp

```diff
@@ -33,6 +33,14 @@ Optional<UISize> Slider::calculated_min_size() const
     size.set_primary_size_for_orientation(orientation(), track_margin() * 2 + knob_fixed_primary_size());
     return size;
 }
+Optional<UISize> Slider::calculated_preferred_size() const
+{
+
+    UISize size { 0, 0 };
+    size.set_secondary_size_for_orientation(orientation(), calculated_min_size()->secondary_size_for_orientation(orientation()));
```

Comment:
Just setting it to `SpecialDimension::Shrink` is the more ideomatic way of achieving this.